### ISSUE TITLE
docs don't build on docs.rs for 1.0.0-alpha.0

### ISSUE THREAD

**Issue by LLFourn** [bitcoindevkit/bdk#955]
See: https://docs.rs/crate/bdk/1.0.0-alpha.0

can we somehow make sure in CI that the docs will build on docs.rs before merging?

**Comment by notmandatory** [comment]
The nightly docs we build in CI are working: https://bitcoindevkit.org/docs-rs/bdk/nightly/latest/bdk/

To troubleshoot why docs.rs docs aren't working we'll need to go through local testing with the docs.rs docker container environment, see: https://github.com/rust-lang/docs.rs/blob/master/README.md

**Comment by notmandatory** [comment]
Adding some notes as I go along incase we run into this issue again.
1. find error page for docs.rs
   https://docs.rs/crate/bdk/1.0.0-alpha.0/builds/771104
2. replicate errors locally
    ```
    cd crates/bdk
    cargo +nightly rustdoc --lib -Zrustdoc-map --all-features -Z unstable-options --config build.rustdocflags='["--cfg", "docsrs", "-Z", "unstable-options", "--emit=invocation-specific", "--resource-suffix", "-20230319-1.70.0-nightly-da7c50c08", "--static-root-path", "/-/rustdoc.static/", "--cap-lints", "warn", "--disable-per-crate-search", "--extern-html-root-takes-precedence"]' -Zunstable-options -Zrustdoc-scrape-examples -j3
    ```

**Comment by notmandatory** [comment]
The problem is we need to enable the `doc_cfg` feature when the `docsrs` configuration attribute is defined:
```rust
#![cfg_attr(docsrs, feature(doc_cfg))]
```
Fix in #1011
